tests/run/typeclass-derivation2a.scala

```diff
@@ -25,12 +25,12 @@ object TypeLevel {
     def mirror(ordinal: Int): Mirror =
       mirror(ordinal, EmptyProduct)
 
-    val label: Array[Array[String]] =
-      initLabels(0, 0, new mutable.ArrayBuffer[String], new mutable.ArrayBuffer[Array[String]])
-
     private final val elemSeparator = '\000'
     private final val caseSeparator = '\001'
```

Comment:
@liufengyun what happens if it is an `inline val`?

Reply:
It should not be an error

Reply:
Yes, due to const-fold it's not an error. The checker now runs before const-fold, it thus reports an error.

But I think it's still good to report errors here, as in this case const-fold changes the semantics of programs, which is too subtle for end users.